src/cookbooks/client_guidelines.md

```diff
@@ -0,0 +1,93 @@
+# Client Implementation Guidelines
+## Feature Gates
+
+The preferred method for testing experimental features is to __land in-tree__ and ship behind a __feature gate__. New features go through the standard Firefox review, testing, and deployment processes, and are then enabled experimentally in the field using these gates.
+A new Feature Gate library for Firefox Desktop is now available. Read the documentation for it to get started.
+
+New features should use this library or an equivalent feature-flagging system as appropriate for their platform.
+
+Historically, Firefox Desktop has used the Preferences system to optionally activate features. __In order for features to be activated experimentally, these rules must be observed__:
+
+It must be possible to enable/disable the feature at runtime, via a pref change.
```

Comment:
Did you want to have bullets here?

Reply:
@harterrt Hm. So, the original doc uses infoboxes... https://docs.google.com/document/d/14y2wTl4ZhFCTwkSRsxwOmjYSPJ0tfhZc6Gw6Qaoa-78/edit#heading=h.hu90n0rnzqaj

I'm not sure if markdown supports this but will look into it (unless you happen to know). Even if it *is* supported, is this something you'd want to see in the docs?

Reply:
I don't think infoboxes are well-supported, at least by most systems that support markdown. I've rewritten a fair bit of the document so these should not be necessary, and used bullet points where appropriate (but would still appreciate review!)